core/App/ehsm_provider.cpp

```diff
@@ -594,6 +617,39 @@ ehsm_status_t AsymmetricDecrypt(ehsm_keyblob_t *cmk,
 
     switch(cmk->metadata.keyspec) {
         case EH_RSA_2048:
+            if (ciphertext->data == NULL || ciphertext->datalen == 0 ||
+                ciphertext->datalen > RSA_OAEP_2048_CIPHER_LENGTH) {
+                return EH_ARGUMENTS_BAD;
+            }
+            /* calculate the ciphertext length */
+            if (plaintext->datalen == 0) {
+                ret = enclave_rsa_decrypt(g_enclave_id,
+                                          &sgxStatus,
+                                          cmk->keyblob,
+                                          cmk->keybloblen,
+                                          ciphertext->data,
+                                          ciphertext->datalen,
+                                          NULL, 0,
+                                          &(plaintext->datalen),
+                                          cmk->metadata.keyspec,
+                                          cmk->metadata.padding_mode);
+                return EH_OK;
+            }
+            /* check if the datalen is valid */
+            if (plaintext->data == NULL || plaintext->datalen == 0)
+                return EH_ARGUMENTS_BAD;
+            ret = enclave_rsa_decrypt(g_enclave_id,
+                                      &sgxStatus,
+                                      cmk->keyblob,
+                                      cmk->keybloblen,
+                                      ciphertext->data,
+                                      ciphertext->datalen,
+                                      plaintext->data,
+                                      plaintext->datalen,
+                                      NULL,
+                                      cmk->metadata.keyspec,
+                                      cmk->metadata.padding_mode);
+            break;
```

Comment:
any difference between the 2048 and 3072?

Reply:
only on line 564, I plan to merge the code for 3072 and 2048